DEVELOPING.md

```diff
@@ -0,0 +1,46 @@
+# Development Guidelines for qrm
+
+**Make sure to also read `CONTRIBUTING.md`, everything in there applies here.**
+
+### Environment Setup
+
+1. [Fork this repo][1] into your own GitHub namespace.
+1. Make sure the `master` branch is up to date, then make yourself a new branch with a descriptive name.
+1. Once the forked repo is cloned and on the proper branch, you can set up the development environment.
+    1. Install python 3.7 or higher.
+    1. Run `make dev-install`.
+       This should install everything you need to develop and run qrm.
+    1. [Create a bot and token][2], and add it to `data/keys.py`.
+       Also add your [QRZ credentials][3] if needed.
+    1. In `data/options.py`, change values as needed.
+       Some commands require adding your Discord user ID to `owner_uids`.
+    1. To activate the virtual env that was created by `make`, run `source botenv/bin/activate` (or the equivelent for your shell or operating system).
+
+## While You Develop
+
+To run qrm, use the command `./run.sh`.
+We recommend you use the `--pass-errors` flags to avoid perpetual restart loops if you break the bot.
+It exists because repeatedly mashing [Ctrl+C] at high speed to break a fast loop is not fun.
+
+Using `dev-notes` for documentation is especially important if you introduce a new json file format or to document some development process (like the command to crush the various images in the repository).
+
+### Test your changes
+
+In addition to testing functionality, make sure to run `flake8` to ensure your code uses the proper style, and `mypy [files...]` to ensure proper typing.
```

Comment:
add 
```md
You can also enable them for this project in your IDE if supported.
This will should give you automatic and continuous linting and type checking.
```